From f018768b3dce49c7de494373b48ccd0ebe4adbe1 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 06 Sep 2022 17:21:27 +0300
Subject: [PATCH] [SW-0] gaudi3: fix local iccm addr translation

local ICCM is accessed via region 0, address translation function
is fixed to give proper address for that region.

Change-Id: I702cb99c3e1e822bd98cb6d2496841cc27e23c83
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 18e9062..ecfbe2d 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -453,7 +453,8 @@
 
 	if (addr >= HL_FW_MEM_BASE_FULL && addr < HL_FW_MEM_BASE_FULL + HL_FW_MEM_SIZE) {
 		if (addr >= ARC1_ICCM_FULL_BASE && addr < ARC1_ICCM_FULL_BASE + ARC1_ICCM_SIZE)
-			return arc_addr; /* access local iccm via mem region 0 */
+			/* access local iccm via mem region 0 */
+			return (addr - ARC1_ICCM_FULL_BASE);
 
 		return (arc_addr + FW_MEM_REGION_BASE);
 	}
